### ISSUE TITLE
keras is very slow with theano

### ISSUE THREAD

**Issue by traveller59** [fchollet/keras#7409]
platform: i5 7500, 1080ti, newest keras and backends, windows 10
Model(MNIST, CNN):
[https://github.com/minimaxir/keras-cntk-benchmark/blob/master/test_files/mnist_cnn.py](url)
model = Sequential()
model.add(Conv2D(20, (5, 5), activation='relu',
                 input_shape=input_shape))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Conv2D(40, (5, 5), activation='relu'))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.5))
model.add(Flatten())
model.add(Dense(128, activation='relu'))
model.add(Dropout(0.5))
model.add(Dense(num_classes, activation='softmax'))

this model takes 2s per epoch with tensorflow, 3s with cntk and 11s with theano.
I have my own simple code to train MNIST and takes 1.3s per epoch with theano, 2.0s with tensorflow and 3.1s with cntk.

**Comment by StefanoD** [comment]
Did you setup theano correctly? In my experiments theano is faster than TF.
Here is my .theanorc config:
```
[global]
floatX=float32
device=cuda0

[nvcc]
flags=-D_FORCE_INLINES

[cuda]
root=/usr/local/cuda/bin

[dnn]
enabled=True

[gpuarray]
preallocate=.90
```

**Comment by traveller59** [comment]
This problem is solved. I changed backend by using “os.environ” before and didn't set "image_data_format" to "channels_first".  Backend should be changed by modifying ~/.keras/keras.json.
